### ISSUE TITLE
[BUG] YouTube cookie consent page breaks chat download process.

### ISSUE THREAD

**Issue by z8i** [xenova/chat-downloader#85]
## Basic information
- Program version: 0.0.8
- Python version: 3.9.2
- Operating system: Windows 10 10.0.19041.867

## Describe the bug
I try to save the chat from some of my videos, but it throws an error:
chat_downloader.errors.ParsingError: Unable to parse initial video data. 

Full Traceback:
chat_downloader https://www.youtube.com/watch?v=x2ukXJM0QO0
[INFO] Site: youtube.com
Traceback (most recent call last):
  File "C:\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\enc\Scripts\chat_downloader.exe\__main__.py", line 7, in <module>
  File "c:\users\user\enc\lib\site-packages\chat_downloader\cli.py", line 170, in main
    run(**args.__dict__)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\chat_downloader.py", line 273, in run
    chat = downloader.get_chat(**chat_params)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\chat_downloader.py", line 166, in get_chat
    info = self.sessions[site.__name__].get_chat(**params)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 1393, in get_chat
    return self.get_chat_by_video_id(match.group('id'), kwargs)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 1367, in get_chat_by_video_id
    initial_info = self._get_initial_video_info(video_id)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 906, in _get_initial_video_info
    raise ParsingError(
chat_downloader.errors.ParsingError: Unable to parse initial video data. 

Followed by a html-page to give cookie consent.
I suspect this is an EU issue...

### If running from the command line, provide the following:
1. The command used (including the verbose tag, `-v`):
```
chat_downloader -v https://www.youtube.com/watch?v=x2ukXJM0QO0
```
2. Output from the above command:
```
[DEBUG] Python version: 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)]
[DEBUG] Program version: 0.0.8
[DEBUG] Initialisation parameters: {'cookies': None, 'proxy': None, 'headers': None}
[INFO] Site: youtube.com
[DEBUG] Program parameters: {'url': 'https://www.youtube.com/watch?v=x2ukXJM0QO0', 'start_time': None, 'end_time': None, 'max_attempts': 15, 'retry_timeout': None, 'timeout': None, 'max_messages': None, 'logging': 'debug', 'pause_on_debug': False, 'exit_on_debug': False, 'message_groups': ['messages'], 'message_types': None, 'format': 'youtube', 'format_file': None, 'chat_type': 'live', 'message_receive_timeout': 0.1, 'buffer_size': 4096, 'inactivity_timeout': None}
[DEBUG] Starting new HTTPS connection (1): www.youtube.com:443
[DEBUG] https://www.youtube.com:443 "GET /watch?v=x2ukXJM0QO0 HTTP/1.1" 302 0
[DEBUG] Starting new HTTPS connection (1): consent.youtube.com:443
[DEBUG] https://consent.youtube.com:443 "GET /m?continue=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3Dx2ukXJM0QO0&gl=DE&m=0&pc=yt&uxe=23983172&hl=en&src=1 HTTP/1.1" 200 None
[DEBUG] Session closed.
Traceback (most recent call last):
  File "C:\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\enc\Scripts\chat_downloader.exe\__main__.py", line 7, in <module>
  File "c:\users\user\enc\lib\site-packages\chat_downloader\cli.py", line 170, in main
    run(**args.__dict__)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\chat_downloader.py", line 273, in run
    chat = downloader.get_chat(**chat_params)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\chat_downloader.py", line 166, in get_chat
    info = self.sessions[site.__name__].get_chat(**params)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 1393, in get_chat
    return self.get_chat_by_video_id(match.group('id'), kwargs)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 1367, in get_chat_by_video_id
    initial_info = self._get_initial_video_info(video_id)
  File "c:\users\user\enc\lib\site-packages\chat_downloader\sites\youtube.py", line 906, in _get_initial_video_info
    raise ParsingError(
chat_downloader.errors.ParsingError: Unable to parse initial video data. 
```
Then the contents of the consent-page, see attachment.
## Expected behavior
Chat-downloader should give consent to the cookie, so i can retrieve the chat messages.
[ytout.txt](https://github.com/xenova/chat-downloader/files/6260319/ytout.txt)

**Comment by 9001** [comment]
for now you can get around it by using youtube-dl to produce a cookies file:
```
# download something from youtube to get the right cookies
python -m youtube_dl --cookies cookies.ytdl https://youtu.be/Z72rgSXpUHU

# replace expiration time 0 with something valid
python3 -c "import time; cookies = open('cookies.ytdl', 'r').read().replace('\t0\t', '\t{:.0f}\t'.format(time.time()+60*60*24*90)); open('cookies.chatdl', 'w').write(cookies)"
```

then run chat-downloader with the additional arguments `-c cookies.chatdl`

**Comment by turbcool** [comment]
There seems to be a working solution https://stackoverflow.com/a/66940841/10369027

**Comment by turbcool** [comment]
I managed to fix it (#88). Solution was found [here](https://stackoverflow.com/a/66940841/10369027).

Until PR is accepted one could use my [fork](https://github.com/turbcool/chat-downloader.git):
- `pip uninstall chat-downloader`
-  in **requirements.txt** replace `chat-downloader==0.0.8`with `git+https://github.com/turbcool/chat-downloader.git#egg=chat-downloader`
- `pip install -r requirements.txt`

**Comment by xenova** [comment]
This has been fixed in the latest version 😃 (v0.1.1).

Let me know if you have any other questions 👍
